src/Helper/Log.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Woocommerce\InfinitePay\Helper;
+
+if ( ! function_exists( 'add_action' ) ) {
+	exit( 0 );
+}
+
+use Woocommerce\InfinitePay\InfinitePayCore;
+
+class Log
+{
+
+	public $log;
+	public $id;
+
+	public function __construct( $payment = null ) {
+		if ( ! empty( $payment ) ) {
+			$this->id = get_class( $payment );
+		}
+
+		return $this->init_log();
+	}
+
+	public function init_log() {
+		if ( class_exists( 'WC_Logger' ) ) {
+			$this->log = new \WC_Logger();
+		} else {
+			$this->log = InfinitePayCore::woocommerce_instance()->logger();
+		}
+		return $this->log;
+	}
+
+	public function write_log( $function, $message ) {
+		$options = get_option('woocommerce_infinitepay_settings');
+
+		//if (issrt($options['enabled_log']) && $options['enabled_log'] == 'yes') {
+			$this->log->add( $this->id, '[' . $function . ']: ' . $message );
+		//}
```

Comment:
This is right?

Reply:
Fixed, the previous plugin version don't respect the enabled choice